storey/flow.py

```diff
@@ -370,6 +382,51 @@ def join_from_response(element, response):
         JoinWithHttp.__init__(self, request_builder, join_from_response, **kwargs)
 
 
+def _build_request_put_records(partition_func, events):
+    record_list_for_json = []
+    for event in events:
+        record = event.element
+        record_as_bytes = None
+        if isinstance(record, bytes):
+            record_as_bytes = record
+        elif isinstance(record, str):
+            record_as_bytes = record.encode("utf-8")
+        else:
+            raise Exception(f'Unsupported record type {type(record)}')
+        record_as_b64_string = str(base64.b64encode(record_as_bytes), "utf-8")
+        record_list_for_json.append({'PartitionKey': partition_func(event), 'Data': record_as_b64_string})
+
+    payload_obj = {
+        'Records': record_list_for_json
+    }
+    return json.dumps(payload_obj)
+
+
+class WriteToV3IOStream(JoinWithHttp, NeedsV3ioAccess):
+
+    def __init__(self, stream_path, partition_func=None, webapi=None, access_key=None, **kwargs):
+        NeedsV3ioAccess.__init__(self, webapi, access_key)
+        if partition_func is None:
+            def f(_):
+                return str(random.random())
+
+            partition_func = f
+        self._partition_func = partition_func
+
+        def request_builder(event):
+            request_body = _build_request_put_records(self._partition_func, [event])
+            return HttpRequest('POST', f'{self._webapi_url}/{stream_path}/', request_body, self._put_records_headers)
+
+        def join_from_response(_, response):
+            if response.status == 200:
+                response_obj = json.loads(response.body)
+                if response_obj['FailedRecordCount'] == 0:
```

Comment:
whathappens if some records failed?

Reply:
`raise V3ioError(f'Failed to put records to V3IO. Got {response.status} response: {response.body}')`